### PR TITLE
CMR 3970 - Functions to interact with bulk update DB tables

### PR DESCRIPTION
**Pull request by laurenfrederick**

### CODE REVIEW

File: ingest-app/src/cmr/ingest/data/bulk_update.clj [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+(defprotocol BulkUpdateStore
+  "Defines a protocol for getting and storing the bulk update status and task-id
+  information."
+  (get-bulk-update-status [db provider-id])
+  (get-bulk-update-task-status [db provider-id task-id])
+  (get-bulk-update-task-collection-status [db task-id])
+  (get-bulk-update-collection-status [db task-id concept-id])
+  (create-and-save-bulk-update-status [db provider-id json-body concept-ids])
+  (update-bulk-update-task-status [db provider-id task-id status status-message])
```
Comment (ygliuvt) [inline ingest-app/src/cmr/ingest/data/bulk_update.clj]: Drop `provider-id` from the arguments.

File: ingest-app/src/cmr/ingest/data/memory_db.clj [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    (reset! (:acl-hash-data-atom this) acl-hash))
+
+  (get-acl-hash
+    [this]
+    (-> this :acl-hash-data-atom deref))
+
+  ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+  bulk-update/BulkUpdateStore
+
+  (get-bulk-update-status
```
Comment (ygliuvt) [inline ingest-app/src/cmr/ingest/data/memory_db.clj]: You will need `reset` function later for testing.

File: ingest-app/src/cmr/ingest/data/memory_db.clj [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    [this provider-id]
+    (some->> (-> this :task-status-atom deref)
```
Comment (ygliuvt) [inline ingest-app/src/cmr/ingest/data/memory_db.clj]: You should be able to call just `@task-status-atom` here.

File: ingest-app/src/cmr/ingest/data/bulk_update.clj [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+   [cmr.common.log :refer (debug info warn error)]
+   [cmr.common.services.errors :as errors]
+   [cmr.common.util :refer [defn-timed] :as util]
+   [cmr.oracle.connection]
+   [cmr.oracle.sql-utils :as su]))
+
+(defprotocol BulkUpdateStore
+  "Defines a protocol for getting and storing the bulk update status and task-id
+  information."
+  (get-bulk-update-status [db provider-id])
```
Comment (laurenfrederick) [inline ingest-app/src/cmr/ingest/data/bulk_update.clj]: Done.

File: ingest-app/src/cmr/ingest/data/bulk_update.clj [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  (get-bulk-update-task-status [db provider-id task-id])
```
Comment (laurenfrederick) [inline ingest-app/src/cmr/ingest/data/bulk_update.clj]: Done.

File: ingest-app/src/cmr/ingest/data/bulk_update.clj (hunk 1 above)
Comment (laurenfrederick) [inline ingest-app/src/cmr/ingest/data/bulk_update.clj]: Done.

File: ingest-app/src/cmr/ingest/data/memory_db.clj (hunk 3 above)
Comment (laurenfrederick) [inline ingest-app/src/cmr/ingest/data/memory_db.clj]: Done.

File: ingest-app/src/cmr/ingest/data/memory_db.clj (hunk 2 above)
Comment (laurenfrederick) [inline ingest-app/src/cmr/ingest/data/memory_db.clj]: Done.

**Pull request merged**